sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLogOptions.java

```diff
@@ -17,7 +18,30 @@
 
     public HttpLogOptions() {
         logLevel = HttpLogDetailLevel.NONE;
-        allowedHeaderNames = new HashSet<>();
+        allowedHeaderNames = new HashSet<>(Arrays.asList(
```

Comment:
Extract the `Arrays.asList()` into a static field and reuse. Don't have to create a list and a hashset each time an instance of HttpLogOptions is created.

Reply:
I partially agree. The Arrays.asList(...) call can be a private static final field, but there should be a new set per instance, so that it can be modified via the API.

Reply:
updated